sky/provision/gcp/instance.py

```diff
@@ -186,28 +186,48 @@ def open_ports(
 
     label_filters = {TAG_RAY_CLUSTER_NAME: cluster_name_on_cloud}
     handlers: List[Type[instance_utils.GCPInstance]] = [
-        instance_utils.GCPComputeInstance
+        instance_utils.GCPComputeInstance,
+        instance_utils.GCPTPUVMInstance,
     ]
     handler_to_instances = _filter_instances(handlers, project_id, zone,
                                              label_filters, lambda _: None)
     operations = collections.defaultdict(list)
+    compute_handler: Type[instance_utils.GCPInstance] = (
+        instance_utils.GCPComputeInstance)
     for handler, instances in handler_to_instances.items():
         if not instances:
             logger.warning(f'No instance found for cluster '
                            f'{cluster_name_on_cloud}.')
             continue
         else:
-            op = handler.create_or_update_firewall_rule(
+            for instance in instances:
+                # Add tags for all nodes in the cluster, so the firewall rule
+                # could correctly apply to all instance in the cluster.
+                handler.add_tag_if_not_exist(
+                    project_id,
+                    zone,
+                    instance,
+                    tag=cluster_name_on_cloud,
+                )
+            # If we have multiple instances, they are in the same cluster,
+            # i.e. the same VPC. So we can just pick any one of them.
+            vpc_name = handler.get_vpc_name(project_id, zone, instances[0])
+            if vpc_name is None:
+                # This is the case if we encountered error when getting
+                # the vpc name.
+                continue
```

Comment:
Why do we skip it instead of raising an error?

Reply:
Just want to make sure, do we want to interrupt the `sky launch` process? The logging information is printed within the `handler.get_vpc_name` function

Reply:
It will fail to open if this is the first time to open ports on this cluster (i.e. if we need to create a new firewall rule) since firewall rule only applies to VMs in the specified VPC

Reply:
Got it! Thanks! 
I think it should be fine to skip the port opening if failure happens, but we should say more about why it is OK to skip in the comment and in what case the error can happen.
```
Failed to get the VPC name for the instance. It may be because xxx. We skip the port opening if the failure happens, because xxx.
```

If we skip the port opening, does that mean later we will still add the ports to the `handle.launched_resources` in the caller?
That said, the user will never be able to open that port, no matter how many times she run the `sky launch`?

Reply:
This error won't happen unless some extremely rare cases happen, like the user terminates the instance manually after provision, before executing this function; or googleapiclient API changes. This was part of the debug code, and now it is mostly a double-check to ensure nothing went wrong.

Reply:
> If we skip the port opening, does that mean later we will still add those ports to the handle.launched_resources in the caller?
Good point! Let me check this and if anything is wrong, we still keep the previous handle's ports. Thanks for pointing that out!

Reply:
> That said, the user will never be able to open that port, no matter how many times she run the sky launch?
If it is due to some flaky bug, maybe retry will help... But the case is not likely to happen

Reply:
If that is rare, I guess it should be fine to just raise an error and ask the user to try again?

Reply:
Got it. Then I guess we could raise in all corner cases to make our code clean. Let me revert it 🫡